people-and-planet-ai/geospatial-classification/e2e_test.py

```diff
@@ -0,0 +1,376 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import platform
+import subprocess
+import time
+import uuid
+import json
+import pandas as pd
+import numpy as np
+
+from google.cloud import aiplatform
+from google.cloud import storage
+import pytest
+import requests
+
+import ee
+import google.auth
+
+credentials, project = google.auth.default()
+ee.Initialize(credentials, project=project)
+
+PYTHON_VERSION = "".join(platform.python_version_tuple()[0:2])
+
+NAME = f"ppai/geospatial-classification-py{PYTHON_VERSION}"
+
+UUID = uuid.uuid4().hex[0:6]
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+
+TIMEOUT_SEC = 30 * 60  # 30 minutes in seconds
+POLL_INTERVAL_SEC = 60  # 1 minute in seconds
+
+VERTEX_AI_SUCCESS_STATE = "PIPELINE_STATE_SUCCEEDED"
+VERTEX_AI_FINISHED_STATE = {
+    "PIPELINE_STATE_SUCCEEDED",
+    "PIPELINE_STATE_FAILED",
+    "PIPELINE_STATE_CANCELLED",
+}
+
+EARTH_ENGINE_SUCCESS_STATE = "SUCCEEDED"
+EARTH_ENGINE_FINISHED_STATE = {"SUCCEEDED"}
+
+BANDS = [
+    "B1",
+    "B2",
+    "B3",
+    "B4",
+    "B5",
+    "B6",
+    "B7",
+    "B8",
+    "B8A",
+    "B9",
+    "B10",
+    "B11",
+    "B12",
+]
+LABEL = "label"
+
+logging.getLogger().setLevel(logging.INFO)
+
+
+@pytest.fixture(scope="session")
+def bucket_name():
+    storage_client = storage.Client()
+
+    bucket_name = f"{NAME.replace('/', '-')}-{UUID}"
+    bucket = storage_client.create_bucket(bucket_name, location=REGION)
+
+    logging.info(f"bucket_name: {bucket_name}")
+    yield bucket_name
+
+    bucket.delete(force=True)
+
+
+@pytest.fixture(scope="session")
+def test_data(bucket_name):
+    label_fc = create_label_fc("labeled_geospatial_data.csv")
+    label_fc = label_fc.map(format_date)
+    data = label_fc.map(get_neighboring_patch).flatten()
+    data = data.randomColumn()
+    data = data.filter(ee.Filter.gte("random", 0.03))
+
+    training_task = ee.batch.Export.table.toCloudStorage(
+        collection=data,
+        description="Training image export: test",
+        bucket=bucket_name,
+        fileNamePrefix="geospatial_training",
+        selectors=BANDS + [LABEL],
+        fileFormat="TFRecord",
+    )
+
+    training_task.start()
+
+    validation_task = ee.batch.Export.table.toCloudStorage(
+        collection=data,
+        description="Validation image export: test",
+        bucket=bucket_name,
+        fileNamePrefix="geospatial_validation",
+        selectors=BANDS + [LABEL],
+        fileFormat="TFRecord",
+    )
+
+    validation_task.start()
+
+    train_status = None
+    val_status = None
+
+    logging.info("Waiting for data export to complete.")
+    for _ in range(0, TIMEOUT_SEC, POLL_INTERVAL_SEC):
+        train_status = ee.data.getOperation(training_task.name)["metadata"]["state"]
```

Comment:
While trying this, apparently we can fetch the status like this as well:

```
train_status = training_task.status()['state']
```

Reply:
Did you try this with pytest? I think I had this earlier and it resulted in a bunch of warnings when I ran the tests that said `getOperation` was the non deprecated way to do this.